### ISSUE TITLE
[Paper Note] Short window attention enables long-term memorization, Loïc Cabannes+, arXiv'25, 2025.09

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#3043]
# URL
- https://arxiv.org/abs/2509.24552
# Authors
  - Loïc Cabannes
  - Maximilian Beck
  - Gergely Szilvasy
  - Matthijs Douze
  - Maria Lomeli
  - Jade Copet
  - Pierre-Emmanuel Mazaré
  - Gabriel Synnaeve
  - Hervé Jégou
# Abstract
  - Recent works show that hybrid architectures combining sliding window softmax attention layers with linear recurrent neural network (RNN) layers outperform both of these architectures taken separately. However, the impact of the window length and the interplay between softmax attention and linear RNN layers remain under-studied. In this work, we introduce SWAX, a hybrid architecture consisting of sliding-window attention and xLSTM linear RNN layers.   A counter-intuitive finding with SWAX is that larger sliding windows do not improve the long-context performance. In fact, short window attention encourages the model to better train the long-term memory of the xLSTM, by relying less on the softmax attention mechanism for long context-retrieval.   The issue with small sliding windows is that they are detrimental for short-context tasks, which could be solved with information from moderately larger sliding windows otherwise. Therefore, we train SWAX by stochastically changing the sliding window size, forcing the model to leverage both a longer context window and the xLSTM memory. SWAX trained with stochastic window sizes significantly outperforms regular window attention both on short and long-context problems.
# Translation (by gpt-4o-mini)
- 最近の研究では、スライディングウィンドウソフトマックスアテンション層と線形再帰神経ネットワーク（RNN）層を組み合わせたハイブリッドアーキテクチャが、これらのアーキテクチャを個別に使用した場合よりも優れた性能を示すことが明らかになっています。しかし、ウィンドウの長さの影響やソフトマックスアテンションと線形RNN層の相互作用については、十分に研究されていません。本研究では、スライディングウィンドウアテンションとxLSTM線形RNN層からなるハイブリッドアーキテクチャSWAXを紹介します。

SWAXに関する直感に反する発見は、より大きなスライディングウィンドウが長いコンテキストのパフォーマンスを向上させないということです。実際、短いウィンドウアテンションは、長期的な記憶をより良く訓練するためにxLSTMに依存し、長いコンテキストの取得においてソフトマックスアテンションメカニズムへの依存を減らすことを促します。

小さなスライディングウィンドウの問題は、短いコンテキストタスクに対して有害であり、そうでなければ適度に大きなスライディングウィンドウからの情報で解決できる可能性があります。したがって、SWAXはスライディングウィンドウのサイズを確率的に変更しながら訓練し、モデルがより長いコンテキストウィンドウとxLSTMメモリの両方を活用するように強制します。確率的なウィンドウサイズで訓練されたSWAXは、短いコンテキストと長いコンテキストの問題の両方において、通常のウィンドウアテンションを大幅に上回る性能を示します。
# Summary (by gpt-4o-mini)
- SWAXというハイブリッドアーキテクチャは、スライディングウィンドウアテンションとxLSTM線形RNN層を組み合わせており、短いウィンドウが長期的な記憶をより良く訓練することを示す。SWAXはウィンドウサイズを確率的に変更し、短い・長いコンテキストの両方で優れた性能を発揮する。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/arankomatsuzaki/status/1972874639333605540?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
openreview:https://openreview.net/forum?id=btgVfhudI1
